File: server/core/model/data_source.go
package model

type DataSource struct {
	BaseModel
	DataSourceName    string `gorm:"column:data_source_name;size:150" json:"dataSourceName"`
	DataSourceType    string `gorm:"column:data_source_type;size:20" json:"dataSourceType"`
	DataSourceDesc    string `gorm:"column:data_source_desc;size:200" json:"dataSourceDesc,omitempty"`
	Address           string `gorm:"column:address;size:200" json:"address"`
	Port              string `gorm:"column:port;size:10" json:"port"`
	UserName          string `gorm:"column:user_name;size:40" json:"userName"`
	Password          string `gorm:"column:password;size:40" json:"password"`
	DatabaseName      string `gorm:"column:database_name;size:40" json:"databaseName"`
	ConnectExtInfo    string `gorm:"column:connect_ext_info;size:200" json:"connectExtInfo,omitempty"`
	MinPoolSize       *int   `gorm:"column:min_pool_size" json:"minPoolSize,omitempty"`
	MaxPoolSize       *int   `gorm:"column:max_pool_size" json:"maxPoolSize,omitempty"`
	QueryTimeout      *int   `gorm:"column:query_timeout" json:"queryTimeout,omitempty"`
	DataSourceExtInfo string `gorm:"column:data_source_ext_info;size:200" json:"dataSourceExtInfo,omitempty"`
}

func (DataSource) TableName() string { return "t_data_source" }

File: server/src/main/java/com/javaoffers/nexus/core/model/DataSource.java
package com.javaoffers.nexus.core.model;

import com.javaoffers.brief.modelhelper.anno.BaseModel;
import com.javaoffers.brief.modelhelper.anno.BaseUnique;
import com.javaoffers.brief.modelhelper.anno.ColName;
import com.javaoffers.brief.modelhelper.anno.derive.flag.IsDel;
import lombok.Data;

import java.util.Date;

@BaseModel("t_data_source")
@Data
public class DataSource {
    @BaseUnique
    private Long id;
    private IsDel deleted;
    private Date createdAt;
    private Long createdBy;
    private Date updatedAt;
    private Long updatedBy;

    @ColName(value = "id", excludeColAll = true)
    private Long countId;

    private String dataSourceName;
    private String dataSourceType;
    private String dataSourceDesc;
    private String address;
    private String port;
    private String userName;
    private String password;
    private String databaseName;
    private String connectExtInfo;
    private Integer minPoolSize;
    private Integer maxPoolSize;
    private Integer queryTimeout;
    private String dataSourceExtInfo;
}
